awx/ui_next/src/screens/Template/shared/SurveyList.jsx

```diff
@@ -1,48 +1,259 @@
-import React, { useEffect, useCallback } from 'react';
+import React, { useEffect, useCallback, useState } from 'react';
 import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
 
-import useRequest from '@util/useRequest';
+import useRequest, { useDeleteItems } from '@util/useRequest';
+import { Switch, Checkbox, Button } from '@patternfly/react-core';
+import {
+  DataToolbar,
+  DataToolbarContent,
+  DataToolbarGroup,
+  DataToolbarItem,
+} from '@patternfly/react-core/dist/umd/experimental';
 import ContentError from '@components/ContentError';
 import ContentLoading from '@components/ContentLoading';
+import ErrorDetail from '@components/ErrorDetail';
 import { JobTemplatesAPI } from '@api';
+import ContentEmpty from '@components/ContentEmpty';
+import { ToolbarAddButton } from '@components/PaginatedDataList';
+import { getQSConfig } from '@util/qs';
 import SurveyListItem from './SurveyListItem';
 
-function SurveyList({ template }) {
+import AlertModal from '../../../components/AlertModal';
+
+const QS_CONFIG = getQSConfig('survey', {
+  page: 1,
+});
+
+function SurveyList({ template, i18n }) {
+  const [selected, setSelected] = useState([]);
+  const [surveyEnabled, setSurveyEnabled] = useState(template.survey_enabled);
+  const [showToggleError, setShowToggleError] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+
   const {
-    result: questions,
+    result: { questions, name, description },
     error: contentError,
     isLoading,
     request: fetchSurvey,
   } = useRequest(
     useCallback(async () => {
       const {
-        data: { spec },
+        data: { spec, description: surveyDescription, name: surveyName },
       } = await JobTemplatesAPI.readSurvey(template.id);
-
-      return spec.map((s, index) => ({ ...s, id: index }));
-    }, [template.id])
+      return {
+        questions: spec?.map((s, index) => ({ ...s, id: index })),
+        description: surveyDescription,
+        name: surveyName,
+      };
+    }, [template.id]),
+    { questions: [], name: '', description: '' }
   );
 
   useEffect(() => {
     fetchSurvey();
   }, [fetchSurvey]);
-  if (contentError) {
-    return <ContentError error={contentError} />;
-  }
-  if (isLoading) {
-    return <ContentLoading />;
-  }
 
-  return (
-    questions?.length > 0 &&
-    questions.map((question, index) => (
+  const isAllSelected =
+    selected.length === questions?.length && selected.length > 0;
+
+  const {
+    isLoading: isDeleteLoading,
+    deleteItems: deleteQuestions,
+    deletionError,
+    clearDeletionError,
+  } = useDeleteItems(
+    useCallback(async () => {
+      if (isAllSelected) {
+        return JobTemplatesAPI.destroySurvey(template.id);
+      }
+      const surveyQuestions = [];
+      questions.forEach(q =>
+        !selected.find(s => s.id === q.id) ? surveyQuestions.push(q) : null
+      );
+      return JobTemplatesAPI.updateSurvey(template.id, {
+        name,
```

Comment:
Why are `name` and `description` values sent for deletion requests?

Reply:
the api requires them...delete is not really the right name for this use case.  It is more of an update.